### PR TITLE
Fix dashboards crashing by making some performance optimizations

### LINKED ISSUE #3711: Loading Dashboard makes browser unresponsive
this dashboard:
[Wireless Signal.json.zip](https://github.com/influxdata/chronograf/files/2113292/Wireless.Signal.json.zip)

when i load it in my chronograf, the browser starts using 100% cpu and the page just crashes.

image

kapture 2018-06-18 at 17 27 21

### PR DESCRIPTION
**Pull request by ischolten**
Co-authored-by: Brandon Farmer <bthesorceror@gmail.com>

Closes #3711 

_What was the problem?_
Lots of performance issues causing Chronograf to crash on dashboard page

_What was the solution?_
A few small optimizations and removing some deep equals comparisons in a componentShouldUpdate() function

  - [x] Rebased/mergeable
  - [x] Tests pass

### CODE REVIEW

File: ui/src/shared/components/Annotations.tsx [hunk 1]
```diff
@@ -92,8 +92,10 @@ class Annotations extends Component<Props> {
   get annotations() {
     return visibleAnnotations(
       this.props.xAxisRange,
-      this.props.annotations
-    ).filter(a => a.id !== TEMP_ANNOTATION.id)
+      this.props.annotations,
+      TEMP_ANNOTATION.id
+    )
+    // .filter(a => a.id !== TEMP_ANNOTATION.id)
```
Comment (121watts) [inline ui/src/shared/components/Annotations.tsx]: comment?

File: ui/src/shared/components/Dygraph.tsx [hunk 2]
```diff
@@ -184,11 +176,11 @@ class Dygraph extends Component<Props, State> {
...
-  public shouldComponentUpdate(nextProps: Props, nextState: State) {
-    const arePropsEqual = _.isEqual(this.props, nextProps)
-    const areStatesEqual = _.isEqual(this.state, nextState)
-    return !arePropsEqual || !areStatesEqual
-  }
+  // public shouldComponentUpdate(nextProps: Props, nextState: State) {
+  //   const arePropsEqual = _.isEqual(this.props, nextProps)
+  //   const areStatesEqual = _.isEqual(this.state, nextState)
+  //   return !arePropsEqual || !areStatesEqual
+  // }
```
Comment (121watts) [inline ui/src/shared/components/Dygraph.tsx]: comment?

File: ui/src/shared/components/Annotations.tsx (hunk 1 above)
Comment (ischolten) [reply ui/src/shared/components/Annotations.tsx]: will delete that

File: ui/src/shared/components/Dygraph.tsx (hunk 2 above)
Comment (ischolten) [reply ui/src/shared/components/Dygraph.tsx]: will delete this too.

**Pull request merged**